snuba/utils/metrics/backends/dummy.py

```diff
@@ -1,9 +1,37 @@
-from collections import Mapping
-from typing import Optional, Union
+from __future__ import annotations
+
+from typing import List, Mapping, MutableMapping, Optional, Tuple, Union
 
 from snuba.utils.metrics.backends.abstract import MetricsBackend
 from snuba.utils.metrics.types import Tags
 
+RecordedMetricCalls = List[Tuple[Union[int, float], Tags]]
```

Comment:
I think this API could be made more user friendly by making this a dataclass.

```python3
@dataclass
class RecordedMetric:
    value: int | float
    tags: Optional[Tags]
```

That way this line:

```
[c for c in metric_calls if c[1]["team"] == "sns"]
```

becomes:

```
[c for c in metric_calls if c.tags["team"] == "sns"]
```

which is much more readable